packages/rum-core/src/domain/assembly.ts

```diff
@@ -88,7 +90,8 @@ export function startRumAssembly(
           date: timeStampNow(),
           service: configuration.service,
           session: {
-            type: sessionType,
+            // must be computed on each event because synthetics instrumentation can be done after sdk execution
```

Comment:
Does that mean that if the first events happen before the synthetics instrumentation, they will be sent with the wrong session type?

Reply:
yes, it was the previous behavior

Reply:
We currently have a RFC that tries to relate RUM events to Synthetics tests, I think some solutions from that RFC could be used for this use case too. In particular, I like the idea of using a `Set-Cookie` header to pass data from Synthetics to the SDK. So we could set the session type based on the presence of a cookie.
Do you think it worth exploring that solution? may be as a follow-up to this PR?

Reply:
We already discussed this topic during our daily yesterday, the solution describe in the RFC could indeed make the synthetics sessions detection more reliable. However, since this solution is not yet available, the goal of this PR is to fix the issue introduced with the previous change.